cf_xarray/accessor.py

```diff
@@ -2744,22 +2745,128 @@ def __getitem__(self, key: Hashable | Iterable[Hashable]) -> DataArray:
 
         return _getitem(self, key)
 
+    def __eq__(self, other: object):
+        """
+        Compare flag values against `other`.
+
+        `other` must be in the 'flag_meanings' attribute.
+        """
+        if other not in self.flags.data_vars:
+            raise ValueError(
+                f"Flag meaning {other!r} not in known meanings "
+                f"{list(self.flags.data_vars)!r}"
+            )
+        return self.flags[other].rename(self._obj.name)
```

Comment:
I thought it was better to preserve the flag variable name since that's consistent with other comparisons

Reply:
Also  I think this is a big slowdown since we now loop over all flag_meanings for a single comparison. 

The old version was just a single pass through the data. Since we know what we're looking for, we should be just doing a single pass over the data.